src/psyclone/psyir/nodes/reference.py

```diff
@@ -274,8 +285,33 @@ def is_full_range(self, index):
                 ScalarType.Intrinsic.INTEGER
                 and lower.children[1].value == str(index+1)):
             return False
+        return True
+
+    def is_upper_bound(self, index):
+        '''Returns True if the array index contains a Range Node that
+        specifies the "stop" part of the (start, stop, step) range
+        values is equal to the largest element in that index, by using
+        the UBOUND(name,index) intrinsic. If this is not the case it
+        returns False.
+
+        For example, if an array A was declared as
+        A(10) then the largest element is 10 and UBOUND(A,1) would
```

Comment:
Pointed out it is Fortran as was done for lbound. I've reworded slightly (to stopping, to match with starting) rather than changing element to index value.